### ISSUE TITLE
SuccessMessageMixin is useless with InlineViews

### ISSUE THREAD

**Issue by pjrobertson** [AndrewIngram/django-extra-views#59]
Using `forms_valid()` and never calling `form_valid()` means that the `SuccessMessageMixin` can't set any success messages. (`form_valid()` is where `SuccessMessageMixin` injects the messages)

To get round this, I made a new `InlineSuccessMessageMixin`

``` python
from django.contrib import messages


class InlineSuccessMessageMixin(object):
    """
    Adds a success message on successful form submission.
    """
    success_message = ''

    def forms_valid(self, form, formset):
        print "something"
        response = super(InlineSuccessMessageMixin, self).forms_valid(form, formset)
        success_message = self.get_success_message(form.cleaned_data)
        if success_message:
            messages.success(self.request, success_message)
        return response

    def get_success_message(self, cleaned_data):
        return self.success_message % cleaned_data
```

It's basically just a copy of the django's built in one.

If you were to implement this, things to note:

It doesn't pass the formset forms' `cleaned_data` to the success message

**Comment by jonashaag** [comment]
This must refer to some old code that doesn't exist anymore. I cant' find `SuccessMessageMixin` in the current code base, nor in any other ticket or pull request. Closing.

**Comment by josechaso** [comment]
If I am understanding @pjrobertson well SuccessMessageMixin is not in this project code, but in django source code:
https://github.com/django/django/blob/master/django/contrib/messages/views.py#L4

I think he is referring to what in django docs is described as the way to add support for messaging to class based views:
https://docs.djangoproject.com/en/1.9/ref/contrib/messages/#adding-messages-in-class-based-views

but adapted to this project formset views. 
I am using this approach (check code below) with ModelFormSetView and it works for me.

Now it should be rewritten like this: 

```
from django.contrib import messages

class InlineSuccessMessageMixin(object):
    """
    Adds a success message on successful form submission.
    """
    success_message = ''

    def formset_valid(self, formset):
        response = super(InlineSuccessMessageMixin, self).formset_valid(formset)
        success_message = self.get_success_message(formset)
        if success_message:
            messages.success(self.request, success_message)
        return response

    def get_success_message(self, formset):
        return self.success_message
```
NOTE: get_success_message method can be customized to read the formset content an build a custom success message using the formset information

**Comment by jonashaag** [comment]
Sounds reasonable. Sorry for closing this so quickly.

PRs welcome.

**Comment by sdolemelipone** [comment]
This is now resolved by #184.
